### PR TITLE
Save and load the current planned hyperjump in the save file

### PR DESCRIPTION
**Pull request by JonBooth78**
The planned hyperspace route, with all stops is saved along with the save game, and restored on load.

### CODE REVIEW

**Comment by impaktor** [comment]
This fixes #5178 ?

**Comment by JonBooth78** [comment]
@impaktor , only half of it; I'll get onto the rest later - thanks for pointing it out

**Review by Web-eWorks — COMMENTED** [review]
Especially given that you're planning on serializing more sector view state, I'd recommend that you marshal the serialized data through a `Serializer:Register()` call in `map-sector-view.lua` instead of registering the serialization callback in `hyperjump-planner.lua`.

The hyperjump planner / sector view split is slightly unconventional / unfinished in that the hyperjump planner is not quite a re-usable class that the sector map owns an instance of (yet), but it's close enough that I'd recommend applying that same mental model.

This is not strictly required, but it will make it easier to clean up / finish encapsulating the hyperjump planner widget in the future.

**Comment by Web-eWorks** [comment]
Hmm... the more I look at this code the more I think the serialization should be entirely handled in `map-sector-view.lua`. 99% of this code is manipulating the C++ sector view object itself, and the only thing that involves the hyperjump planner widget is a one-off call in an onGameStart event to `hyperjumpPlanner.buildRouteList()`.

**Comment by JonBooth78** [comment]
@Web-eWorks , I think your right; I also suspect that the call to `hyperjumpPlanner.buildRouteList()` is also redundant as I think it gets called on entry to the screen; I just wanted to be super-safe.

I felt that as this was all to do with the route-planner, it made sense to live in here, although I agree, it really has little to do with that element, which is a UX one.

I'll have a crack at moving it to the suggested place and add the other missing bits of #5187

**Comment by JonBooth78** [comment]
@Web-eWorks , ready for re-review, all seems to be complete and working, with a bonus bug-fix thrown in at the end :-)

**Comment by JonBooth78** [comment]
oh snap, take that back found a bug

**Comment by JonBooth78** [comment]
OK, fixed that bug (doesn't look like I introduced it, oddly) and then clang formatted a file I touched on the way through. @Web-eWorks now ready to go

File: data/pigui/views/map-sector-view.lua
Comment (Web-eWorks) [inline data/pigui/views/map-sector-view.lua:20]: Teensy tiny nitpick but we generally prefer to require "core" modules like Serializer, Game, Event etc. at the top of the file before other requires. In this case, I'd move the `require 'Serializer'` statement to just after (or before, if you want to sort semi-alphabetically) `require 'SystemPath'`.

File: src/SectorView.cpp
Comment (Web-eWorks) [inline src/SectorView.cpp:29]: Since clang-format is already nitpicking this, I'd move these includes down by one line so they're contiguous with the `#include <unordered_set>` statement and separated from `#include "matrix4x4.h".

**Review by Web-eWorks — APPROVED** [review]
Two very minor tiny nitpicks but the code looks good otherwise. Haven't had time to test this branch, so it's provisionally approved pending a playtest sometime in the next 24 hours.

**Comment by Web-eWorks** [comment]
> OK, fixed that bug (doesn't look like I introduced it, oddly) and then clang formatted a file I touched on the way through.

This is quite the ancient bug oddly enough (4 years old!) but was completely benign up until yesterday when I merged #5724. I had the feeling something was odd with the `onEnterSystem` callback signature but rationalized it away due to lack of time. Thanks for catching that!

**Pull request merged**
